tapdb/ignore_tree.go

```diff
@@ -0,0 +1,380 @@
+package tapdb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/mssmt"
+	"github.com/lightninglabs/taproot-assets/universe"
+
+	lfn "github.com/lightningnetwork/lnd/fn/v2"
+)
+
+// IgnoreUniverseTree is a structure that holds the DB for ignore operations.
+type IgnoreUniverseTree struct {
+	db BatchedUniverseTree
+}
+
+// NewIgnoreUniverseTree returns a new IgnoreUniverseTree with the target DB.
+func NewIgnoreUniverseTree(db BatchedUniverseTree) *IgnoreUniverseTree {
+	return &IgnoreUniverseTree{db: db}
+}
+
+// specifierToIdentifier converts an asset.Specifier into a universe.Identifier.
+func specifierToIdentifier(spec asset.Specifier) (universe.Identifier, error) {
+	var id universe.Identifier
+
+	// The specifier must have a group key to be able to be used within the
+	// ignore tree context.
+	if !spec.HasGroupPubKey() {
+		return id, fmt.Errorf("group key must be set")
+	}
+
+	id.GroupKey = spec.UnwrapGroupKeyToPtr()
+
+	id.ProofType = universe.ProofTypeIgnore
+	return id, nil
+}
+
+// AddTuple adds a new ignore tuples to the ignore tree.
+func (it *IgnoreUniverseTree) AddTuples(ctx context.Context,
+	spec asset.Specifier, tuples ...*universe.SignedIgnoreTuple,
+) lfn.Result[[]universe.AuthenticatedIgnoreTuple] {
```

Comment:
Should we create a local alias `IgnoreTuples = []universe.AuthenticatedIgnoreTuple` to shorten some of these definitions?